web/js/layers/wv.layers.modal.js

```diff
@@ -57,54 +58,117 @@ wv.layers.modal = wv.layers.modal || function(models, ui, config) {
   };
 
   // Create container for 'by interest' filters buttons
-  var $nav = $('<nav></nav>')
-    .attr('id', 'categories-nav');
+  var $nav = $('<nav />', {
+    id: 'categories-nav'
+  });
 
   // Create container for breadcrumb
-  var $breadcrumb = $('<nav></nav>')
-    .attr('id', 'category-breadcrumb');
+  var $breadcrumb = $('<nav />', {
+    id: 'category-breadcrumb'
+  });
+
+  /**
+   * var hasMeasurementSetting - Checks the (current) measurement's source
+   *  for a setting and returns true if present.
+   *
+   * @param  {type} current The current config.measurements measurement.
+   * @param  {type} source  The current measurement source.
+   * @return {type}         Return true if the source contains settings.
+   *
+   */
+  var hasMeasurementSetting = function(current, source) {
+    var projection = models.proj.selected.id;
+    var hasSetting;
+    Object.values(source.settings).forEach(function(setting) {
+      var layer = config.layers[setting];
+      if(layer) {
+        var proj = layer.projections;
+        if(layer.id == setting && Object.keys(proj).indexOf(projection) > -1) {
+          if (layer.layergroup && layer.layergroup.indexOf("reference_orbits") !== -1) {
+            if(current.id === "orbital-track") {
+              hasSetting = true;
+            }
+          // Don't output sources with only orbit tracks
+          } else {
+            hasSetting = true;
+          }
+        }
+      }
+    });
+    return hasSetting ? true : false;
+  };
+
+  /**
+   * var hasMeasurementSource - Checks each (current) measurement's sources
+   *  and run hasMeasurementSetting to see if these sources contain settings.
+   *  If a source contains settings, also sets a hasMeasurement flag to be checked
+   *  when drawing categories.
+   *
+   * @param  {type} current The current config.measurements measurement.
+   * @return {type}         Return true if the measurement has sources with settings.
+   */
+
+  var hasMeasurementSource = function(current) {
+    var hasSource;
+    Object.values(current.sources).forEach(function(source) {
+      if(hasMeasurementSetting(current, source)) { hasSource = true; hasMeasurement = true; };
```

Comment:
I think `hasMeasurement` is unused, so I'd simplify to;
`hasSource = hasMeasurementSetting(current, source);`